### ISSUE TITLE
v13 Processing of bounce emails fails for duplicated mass mailings

### ISSUE THREAD

**Issue by CratlePeter** [odoo/odoo#60761]
Impacted versions: 13

Steps to reproduce: Create and schedule a mass mailing that targets an email address that doesn't exist.  Duplicate that mailing and change the recipient list to another contact who's email address doesn't exist.

Current behavior: For the first mailing, the email is correctly marked as having bounced.  The bounce is not logged against the the mass mailing that's the duplicate.

Expected behavior: A duplicated mailing should only duplicate the mailing message and settings and should act otherwise as a new mailing.  Any bounces should be correctly logged against the duplicate.

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
But do you mean the second mailing list is sent, and then bounced?

**Comment by CratlePeter** [comment]
Yes, that's correct but I'm using contacts, not mailing lists.  To clarify: the duplicate mass mailing is targetted at another contact whose email address that doesn't exist, scheduled and sent.

**Comment by pedrobaeza** [comment]
Well, I think you are twisting a bit the contact option, as that's can't been handled separately, but I don't fully understand the use case. I tag for any other giving more clues.

**Comment by CratlePeter** [comment]
The use case is quite easy: I've created the perfectly formatted and worded mass mailing email and I want to send that marketing email to 50 contacts.  Some of the email addresses will bounce because they've written it down incorrectly or given an invalid email address.  Odoo correctly marks any of the 50 that bounce as bounced.

Next week, when I get another 25 contacts to send a marketing email to, I want to duplicate the original mailing so that I don't have to create the perfectly formatted and worded email again.  When I send the mailing again, some of the new 25 contact email addresses will bounce because they've written it down incorrectly or given an invalid email address.  I need the bounces for the 25 to be logged correctly.

**Comment by pedrobaeza** [comment]
But if they are the same previously bounced, AFAIK they are not sent again. Or are they new incorrect mails?

**Comment by CratlePeter** [comment]
They are new incorrect email addresses.  A new set of people to whom I want to send the same perfectly formatted email.  The email addresses in the sets of 25 and the 50 are mutually exclusive.

**Comment by pedrobaeza** [comment]
OK, very weird that they don't get the bounce status, but maybe an internal token/something is not reset on duplication.

@tde-banana-odoo do you figure out something?

**Comment by tde-banana-odoo** [comment]
Hello,

Blacklist records have a message_bounce counter field (number of times it bounced). As this is an integer, it is probably copied when duplicating records ... and it should probably not.

But looking at @CratlePeter it seems to be something else event if I don't understand the whole use case. You say that some of your "new incorrect addresses" are not sent ? Or not bounced ? Do you have access to server logs as mailgateway gives some logs when dealing with incoming emails ?

Thanks,

**Comment by CratlePeter** [comment]
The email to the "new incorrect addresses" is sent.  They are bounced back.  I can see the bounce in the incoming mail server logs.  It logs as 

`delivered to command: /etc/postfix/odoo-mailgate.py -d odo -u X -p <password> --host <IP address> --port 8069`

Odoo logs:

```
2020-10-26 12:09:43,475 7000 INFO odo odoo.addons.mail.models.mail_thread: Routing mail from mailer-daemon@talktalk.net to bounce+1836-res.partner-2588@c
ratle.co.uk with Message-Id <20201026120942.ED1653DFA4@cratlesupport.cratle.co.uk>: not routing bounce email from cratlelimied@gmail.com replying to ['<1
10136437127352.1603714180.482595443725586-openerp-reply_to@amdtc>'] (no document found)
```
